_lib: Test error in test_warning_calls_filters because of a Python 3.12 deprecation

I get the following error when I run the tests with Python 3.12 on Linux with SciPy `'1.12.0.dev0+2038.f34c611'`.

Short summary:

    ERROR scipy/_lib/tests/test_warnings.py::test_warning_calls_filters - DeprecationWarning: Attribute s is deprecated and will be removed in Python 3.14; use value instead

<details>
<summary>
Full error log
</summary>

```
____________________________ ERROR at setup of test_warning_calls_filters _____________________________
[gw8] linux -- Python 3.12.0 /home/warren/py3.12.0/bin/python3
scipy/_lib/tests/test_warnings.py:76: in warning_calls
    finder.visit(tree)
        bad_filters = []
        bad_stacklevels = ['__init__.py:74', 'ndimage/_interpolation.py:619', 'ndimage/_interpolation.py:849', 'ndimage/_ni_support.py:86', 'signal/_signaltools.py:1564']
        base       = PosixPath('/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy')
        file       = <_io.TextIOWrapper name='/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy/signal/_ltisys.py' mode='r' encoding='utf-8'>
        finder     = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        path       = PosixPath('/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy/signal/_ltisys.py')
        tree       = <ast.Module object at 0x7f72ff7aff90>
/home/warren/py3.12.0/lib/python3.12/ast.py:407: in visit
    return visitor(node)
        method     = 'visit_Module'
        node       = <ast.Module object at 0x7f72ff7aff90>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        visitor    = <bound method NodeVisitor.generic_visit of <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>>
/home/warren/py3.12.0/lib/python3.12/ast.py:415: in generic_visit
    self.visit(item)
        field      = 'body'
        item       = <ast.FunctionDef object at 0x7f72ff8bf250>
        node       = <ast.Module object at 0x7f72ff7aff90>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        value      = [<ast.Expr object at 0x7f72ff7affd0>, <ast.Import object at 0x7f72ff7aff10>, <ast.ImportFrom object at 0x7f72ff7aeb50>...m object at 0x7f72ff7aef50>, <ast.ImportFrom object at 0x7f72ff7aee50>, <ast.ImportFrom object at 0x7f72ff7aedd0>, ...]
/home/warren/py3.12.0/lib/python3.12/ast.py:407: in visit
    return visitor(node)
        method     = 'visit_FunctionDef'
        node       = <ast.FunctionDef object at 0x7f72ff8bf250>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        visitor    = <bound method NodeVisitor.generic_visit of <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>>
/home/warren/py3.12.0/lib/python3.12/ast.py:415: in generic_visit
    self.visit(item)
        field      = 'body'
        item       = <ast.With object at 0x7f72ff93b350>
        node       = <ast.FunctionDef object at 0x7f72ff8bf250>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        value      = [<ast.Expr object at 0x7f72ff8bf8d0>, <ast.Expr object at 0x7f72ff8bf9d0>, <ast.If object at 0x7f72ff8bfe50>, <ast.Assign object at 0x7f72ff938e90>, <ast.If object at 0x7f72ff9399d0>, <ast.Assign object at 0x7f72ff939590>, ...]
/home/warren/py3.12.0/lib/python3.12/ast.py:407: in visit
    return visitor(node)
        method     = 'visit_With'
        node       = <ast.With object at 0x7f72ff93b350>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        visitor    = <bound method NodeVisitor.generic_visit of <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>>
/home/warren/py3.12.0/lib/python3.12/ast.py:415: in generic_visit
    self.visit(item)
        field      = 'body'
        item       = <ast.Expr object at 0x7f72ff93bfd0>
        node       = <ast.With object at 0x7f72ff93b350>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        value      = [<ast.Expr object at 0x7f72ff93bfd0>, <ast.Assign object at 0x7f72ff93b090>]
/home/warren/py3.12.0/lib/python3.12/ast.py:407: in visit
    return visitor(node)
        method     = 'visit_Expr'
        node       = <ast.Expr object at 0x7f72ff93bfd0>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        visitor    = <bound method NodeVisitor.generic_visit of <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>>
/home/warren/py3.12.0/lib/python3.12/ast.py:417: in generic_visit
    self.visit(value)
        field      = 'value'
        node       = <ast.Expr object at 0x7f72ff93bfd0>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        value      = <ast.Call object at 0x7f72ff93bf50>
/home/warren/py3.12.0/lib/python3.12/ast.py:407: in visit
    return visitor(node)
        method     = 'visit_Call'
        node       = <ast.Call object at 0x7f72ff93bf50>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
        visitor    = <bound method FindFuncs.visit_Call of <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>>
scipy/_lib/tests/test_warnings.py:42: in visit_Call
    if node.args[0].s == "ignore":
        node       = <ast.Call object at 0x7f72ff93bf50>
        p          = <scipy._lib.tests.test_warnings.ParseCall object at 0x7f72ffd1fb00>
        self       = <scipy._lib.tests.test_warnings.FindFuncs object at 0x7f7408cefa10>
/home/warren/py3.12.0/lib/python3.12/ast.py:532: in _s_getter
    warnings._deprecated(
        self       = <ast.Constant object at 0x7f72ff93bd50>
        warnings   = <module 'warnings' from '/home/warren/py3.12.0/lib/python3.12/warnings.py'>
/home/warren/py3.12.0/lib/python3.12/warnings.py:529: in _deprecated
    warn(msg, DeprecationWarning, stacklevel=3)
E   DeprecationWarning: Attribute s is deprecated and will be removed in Python 3.14; use value instead
        _version   = sys.version_info(major=3, minor=12, micro=0, releaselevel='final', serial=0)
        message    = '{name} is deprecated and will be removed in Python {remove}; use value instead'
        msg        = 'Attribute s is deprecated and will be removed in Python 3.14; use value instead'
        name       = 'Attribute s'
        remove     = (3, 14)
        remove_formatted = '3.14'
```

</details>

I can fix this locally by changing this line 

https://github.com/scipy/scipy/blob/f34c611eb901ba452a28f4f8b039b79bac5e13d7/scipy/_lib/tests/test_warnings.py#L42

to

            if node.args[0].value == "ignore":

but I haven't looked into ensuring that the change works for older supported versions of Python.

--- a/scipy/_lib/tests/test_warnings.py
+++ b/scipy/_lib/tests/test_warnings.py
@@ -39,7 +39,7 @@ class FindFuncs(ast.NodeVisitor):
         ast.NodeVisitor.generic_visit(self, node)
 
         if p.ls[-1] == 'simplefilter' or p.ls[-1] == 'filterwarnings':
-            if node.args[0].s == "ignore":
+            if node.args[0].value == "ignore":
                 self.bad_filters.append(
                     f"{self.__filename}:{node.lineno}")
 

